Document management scripts usage (#1389)
kyma-project/console: README.md

2. Install Console dependencies. To install dependencies for the root and all UI projects, and prepare symlinks for local libraries within this repository, run the following command:

   ```bash
   npm run bootstrap
   ```

   > **NOTE:** The `npm run bootstrap` command:
   >
   > - installs root dependencies provided in the [package.json](./package.json) file
   > - installs dependencies for the [`React common`](./common), [`React components`](./components/react), [`Shared components`](./components/shared) and [`Generic documentation`](./components/generic-documentation) libraries
   > - builds all the libraries
   > - installs dependencies for all the [components](#components)
   > - updates your `/etc/hosts` with the `127.0.0.1 console-dev.kyma.local` host
   > - creates the `.clusterConfig.gen` file if it doesn't exist, pointing at the `kyma.local` domain

## Usage

### Set the cluster (optional)

> By default, the Kyma cluster URL (which the Console communicates with) is set to `kyma.local`. You can set it to any other address by executing

Reviewer: ```suggestion
By default, the Kyma cluster URL with which the Console communicates is set to `kyma.local`. To change the address of the cluster, run: 
```

```bash

> ./scripts/.setClusterConfig <cluster_url>

Reviewer: ```suggestion
./scripts/.setClusterConfig {CLUSTER_URL}
```

```

To simplify switching clusters hosted on the same domain, you can assign the domain to `CLUSTER_HOST` environment variable, then use any subdomain as a cluster name.

> For example, let's assume you have two clusters you want to easily switch between - `foo.abc.com` and `bar.abc.com`. You can do the following:

Reviewer: ```suggestion
For example, let's assume you want to easily switch between two clusters - `foo.abc.com` and `bar.abc.com`. Follow these steps to simplify switching between these clusters: 
```

```bash

> export CLUSTER_HOST=abc.com # setting it permanently in .bashrc would be even better

Reviewer: ```suggestion
export CLUSTER_HOST=abc.com 
# If you use only one domain for your cluster, consider setting it permanently in your shell.
```
Not everybody uses bash, right? ;)

./scripts/.setClusterConfig foo

> # will do the same as ./scripts/.setClusterConfig foo.abc.com

Reviewer: ```suggestion
# After setting the CLUSTER_HOST variable this is equal to running ./scripts/.setClusterConfig foo.abc.com
```

> # and then, to swithch the cluster:

Reviewer: Remove this line pls

Author: I wanted to mark these two scripts are not meant to be executed one after another.

./scripts/.setClusterConfig bar
# will do the same as ./scripts/.setClusterConfig bar.abc.com
```

You can always set the domain back to `kyma.local` via

```bash
./scripts/.setClusterConfig local
```

### Start all views

Use the following command to run the Console with the [`core`](./core) and all other views locally:

```bash
npm run start
```

> To access the local instance of the Console at `http://console-dev.kyma.local:4200`, use credentials from [this](https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-a-cluster--provider-installation--aks--access-the-cluster) document - point 3.

Reviewer: ```suggestion
To get the credentials required to access the local instance of the Kyma Console at `http://console-dev.kyma.local:4200`, follow the instructions from [this](https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-a-cluster-access-the-cluster) document.
```

### Development with local GraphQL API

> By default, the [`core`](./core) and all other views are connected to the **GraphQL API** running on the cluster at the `https://console-backend.<kyma_domain>/graphql` address. If you want to use local **GraphQL API** endpoint, follow the instructions in the **Run a local version** section of [this](https://github.com/kyma-project/kyma/tree/master/components/console-backend-service#run-a-local-version) document and run the following command:

Reviewer: ```suggestion
By default, the [`core`](./core) view and all other views are connected to the **GraphQL API** running on the cluster at the `https://console-backend.{CLUSTER_DOMAIN}/graphql` address. If you want to use the local **GraphQL API** endpoint, follow the instructions in the **Run a local version** section of [this](https://github.com/kyma-project/kyma/tree/master/components/console-backend-service#run-a-local-version) document and run this command:
```

### Can't access `console.kyma.local` and `console-dev.kyma.local:4200` after hibernating the Minikube cluster

Follow the guidelines from [this](https://kyma-project.io/docs/master/root/kyma/#troubleshooting-basic-troubleshooting-can-t-log-in-to-the-console-after-hibernating-the-minikube-cluster) document to solve the problem.

### Check the availability of a remote cluster

Use the `checkClusterAvailability.sh` script to quickly check the availability of remote clusters.

```bash
./scripts/checkClusterAvailability.sh {CLUSTER_URL}

# or

export CLUSTER_HOST=abc.com
./scripts/checkClusterAvailability.sh <cluster_subdomain>
# the same as ./scripts/checkClusterAvailability.sh {CLUSTER_SUBDOMAIN}.abc.com

# or

./scripts/checkClusterAvailability.sh
# Checks the availability of every cluster that has ever been set through setClusterConfig.sh or checked with checkClusterAvailability.sh on your machine.

# or

./scripts/checkClusterAvailability.sh -s <cluster_domain>

> # will sillently return the exit code 0 or another if the cluster is unavailable

Reviewer: ```suggestion
# Returns an appropriate exit code if the cluster is unavailable.
```

### Can't access `console.kyma.local` and `console-dev.kyma.local:4200` after hibernating the Minikube cluster

Follow the guidelines from [this](https://kyma-project.io/docs/master/root/kyma/#troubleshooting-basic-troubleshooting-can-t-log-in-to-the-console-after-hibernating-the-minikube-cluster) document to solve the problem.

### I'd like to quickly check if my remote cluster is running

> There's a script that let's you do this with one command.

Reviewer: ```suggestion
Use the `checkClusterAvailability.sh` script to quickly check the availability of remote clusters.
```

```bash

> ./scripts/checkClusterAvailability.sh <cluster_url>

Reviewer: ```suggestion
./scripts/checkClusterAvailability.sh {CLUSTER_URL}
```

# or

export CLUSTER_HOST=abc.com
./scripts/checkClusterAvailability.sh <cluster_subdomain>
# the same as ./scripts/checkClusterAvailability.sh <cluster_subdomain>.abc.com

# or

./scripts/checkClusterAvailability.sh

> # will check every cluster address that has ever been set (via setClusterConfig.sh) or checked (via checkClusterAvailability.sh) on your machine

Reviewer: ```suggestion
# Checks the availability of every cluster that has ever been set through setClusterConfig.sh or checked with checkClusterAvailability.sh on your machine.
```

# or

./scripts/checkClusterAvailability.sh -s <cluster_domain>

> # will sillently return the exit code 0 or another if the cluster is unavailable

Reviewer: So what does it return? "Or another" is pretty enigmatic.

Author: It returns the return code of the `nc` command which is inside the script and actually checks the endpoint availability. This means 0 if the cluster is available or other code it it isn't (the actual code depends on which problem has `nc` encountered - domain not resolved, not responding, some server error etc.).
